### PR TITLE
Rebrand project to Omni Core

### LINKED ISSUE #10: Rebranding / file names
In earlier versions `bitcoind`, `bitcoin-cli` and `bitcoin-qt` were renamed to `mastercored`, `mastercore-cli` and `mastercore-qt` via makefile.

In https://github.com/OmniLayer/omnicore/pull/8#issuecomment-86921420 @m21 tackled the topic and suggested renaming the files to `omnicore*`.

While I generally agree with this practise, there are two points to discuss:

1) The next release:

I'm in the camp that favors a minor release, based on Bitcoin Core 0.10, before doing a major release with meta DEx and new encoding scheme, because roughly 40 % of the nodes listed on [bitnodes.io](https://getaddr.bitnodes.io/nodes/?q=/Satoshi:0.10.0/) already use 0.10 and such a release is basically a byproduct of the progress towards the next major release. If this is done, the files names should probably still be `mastercore*` instead of `omnicore*`, for a frictionless integration.

2) The issues related to file renaming:

There are several parts, which refer to `bitcoin*` files, for example the [QA tests](https://github.com/OmniLayer/omnicore/tree/omnicore-0.0.10/qa), the deterministic building via Gitian, the Windows installer (if I'm not mistaken, handled by [share/setup.nsi.in](https://github.com/OmniLayer/omnicore/blob/omnicore-0.0.10/share/setup.nsi.in)), and probably other parts as well.

For the last release a [new branch](https://github.com/mastercoin-MSC/mastercore/tree/mscore-0.0.9-gitianfix) was even created to revert the file names as workaround.

- It is thinkable to extend the scope of the rebranding and update all `bitcoin*` file references. This is probably the way to go, but as natural consequence, this can result in compatibility issues and merge conflicts with Bitcoin Core.

- Alternatively the build route could be adjusted to produce `bitcoin*` and `omnicore*` files at the same time.

- Or there could be an option to define the file name target via `./configure`. This would probably provide a maximum of flexibility.

I'm not sure, how to tackle this, but in my opinion the rebanding should not break things, and I'd be much more comfortable, if there were a strategy that considers all aspects, whether this involves an extended scope or a dual building route, or something else.

### LINKED ISSUE #97: Prepare installers, replace icons, define version
With #91, at least for Windows, the setup and icons currently like this:

qtinst

It should be rebranded to Omni Core.

- [x] Prepare installer
- [x] Replace setup icons
- [x] Define version

### PR DESCRIPTION
**Pull request by dexX7**
This PR rebrands the project to Omni Core.

It updates file names, including setup packages. I haven't tested it cross-plattform, and neither the installers.

Given that Travis failed a few times, due to some mistakes, I'm somewhat confident that it's going to work though. Nevertheless, testing should be done rather sooner than later, with special attention to default datadirs, install locations and more importantly: deinstallation routines.

It resolves #10, and it resolves #97.

Edit:

I added resource files for the Windows binaries: [src/omnicore/res](https://github.com/dexX7/bitcoin/tree/30fdde4833ac20c130d62e9d560b3b0d70ee8744/src/omnicore/res), and started with the documentation [src/omnicore/doc](https://github.com/dexX7/bitcoin/tree/30fdde4833ac20c130d62e9d560b3b0d70ee8744/src/omnicore/doc). This isn't intended to finalize the docs, but rather to have the file structure for the setup script.

The Omni Core version needs to be defined in [configure.ac](https://github.com/dexX7/bitcoin/blob/30fdde4833ac20c130d62e9d560b3b0d70ee8744/configure.ac#L9-L13) as well as in [src/omnicore/version.h](https://github.com/dexX7/bitcoin/blob/30fdde4833ac20c130d62e9d560b3b0d70ee8744/src/omnicore/version.h#L14-L27). The values of the configure file are used to replace placeholder values in other files like the setup script.

The version is still `"0.0.10.0-dev"`, and I called the very first digit "milestone". This is for no particular reason, but it needed to be defined somehow.

When building, we end up with the following binaries:

- `src/omnicore-cli`
- `src/omnicored`
- `src/test/test_omnicore`
- `src/qt/omnicore-qt`
- `src/qt/test/test_omnicore-qt`

The installer for Windows is currently build with the file name `omnicore-0.0.10.0-dev-win64-setup.exe`, and this is based on the values used in the configuration file.

When using the Windows installer, the following files are created or copied to the install destination:

- `daemon\omnicored.exe`
- `daemon\omnicore-cli.exe`
- `doc\*.*`
- `copying.txt`
- `omnicore-qt.exe`
- `readme.txt`
- `release-notes.txt`
- `uninstall.exe`

This is similar to the original Bitcoin Core behavior, except that it uses Omni Core related documents.

On OS X it *should* be `OmniCore-Qt.app`. I haven't tested it yet, but I also haven't touched that part in the recent update of this PR.

### CODE REVIEW

**Comment by dexX7** [comment]
On hold until I tested the results.

**Comment by dexX7** [comment]
Windows works fine! :)

qtinst

To do:

- replace installer icon
- replace setup images
- define what goes into the package (right now the release notes, and a few other docs are copied)

**Comment by zathras-crypto** [comment]
Nice, nice, nice.  Every day I get up to such great work from you bud :)

Re installation in Ubuntu, is there even an installer for Ubuntu?  I thought they were delivered as straight bins?  

Re installation in OSX, if you have an installer you'd like me to try out more than happy to :)

Do you need any help with the images?

**Comment by dexX7** [comment]
Thanks! :)

> Re installation in Ubuntu, is there even an installer for Ubuntu? I thought they were delivered as straight bins?

Ah, right. Let me rephrase: I haven't tested the build route for Unix.

> Re installation in OSX, if you have an installer you'd like me to try out more than happy to :)

These are the results of the earlier builds:

- http://bitwatch.co/uploads/builds/2015-06-05-gitian-osx-win.rar

Please ping me, once you got them. :)

> Do you need any help with the images?

That would be awesome!

[share/setup.nsi.in](https://github.com/OmniLayer/omnicore/blob/omnicore-0.0.10/share/setup.nsi.in) describes the setup on Windows, and this section sets the icons and images:
```
# MUI Symbol Definitions
!define MUI_ICON "@abs_top_srcdir@/share/pixmaps/bitcoin.ico"
!define MUI_WELCOMEFINISHPAGE_BITMAP "@abs_top_srcdir@/share/pixmaps/nsis-wizard.bmp"
!define MUI_HEADERIMAGE
!define MUI_HEADERIMAGE_RIGHT
!define MUI_HEADERIMAGE_BITMAP "@abs_top_srcdir@/share/pixmaps/nsis-header.bmp"
!define MUI_FINISHPAGE_NOAUTOCLOSE
!define MUI_STARTMENUPAGE_REGISTRY_ROOT HKLM
!define MUI_STARTMENUPAGE_REGISTRY_KEY ${REGKEY}
!define MUI_STARTMENUPAGE_REGISTRY_VALUENAME StartMenuGroup
!define MUI_STARTMENUPAGE_DEFAULTFOLDER "@PACKAGE_NAME@"
!define MUI_FINISHPAGE_RUN $INSTDIR\bitcoin-qt.exe
!define MUI_UNICON "${NSISDIR}\Contrib\Graphics\Icons\modern-uninstall.ico"
!define MUI_UNWELCOMEFINISHPAGE_BITMAP "@abs_top_srcdir@/share/pixmaps/nsis-wizard.bmp"
!define MUI_UNFINISHPAGE_NOAUTOCLOSE
```

So these are the ones we should replace:
```
- share/pixmaps/bitcoin.ico
- pixmaps/nsis-header.bmp
- pixmaps/nsis-wizard.bmp
```

**Comment by zathras-crypto** [comment]
> Please ping me, once you got them. :)

Got them :)

**Comment by zathras-crypto** [comment]
> So these are the ones we should replace

How about these? https://github.com/OmniLayer/omnicore/pull/102

**Comment by dexX7** [comment]
Prr.. this PR wasn't fun.

I added resource files for the Windows binaries: [src/omnicore/res](https://github.com/dexX7/bitcoin/tree/30fdde4833ac20c130d62e9d560b3b0d70ee8744/src/omnicore/res), and started with the documentation [src/omnicore/doc](https://github.com/dexX7/bitcoin/tree/30fdde4833ac20c130d62e9d560b3b0d70ee8744/src/omnicore/doc). This isn't intended to finalize the docs, but rather to have the file structure for the setup script.

The Omni Core version needs to be defined in [configure.ac](https://github.com/dexX7/bitcoin/blob/30fdde4833ac20c130d62e9d560b3b0d70ee8744/configure.ac#L9-L13) as well as in [src/omnicore/version.h](https://github.com/dexX7/bitcoin/blob/30fdde4833ac20c130d62e9d560b3b0d70ee8744/src/omnicore/version.h#L14-L27). The values of the configure file are used to replace placeholder values in other files like the setup script.

The version is still `"0.0.10.0-dev"`, and I called the very first digit "milestone". This is for no particular reason, but it needed to be defined somehow.

When building, we end up with the following binaries:

- `src/omnicore-cli`
- `src/omnicored`
- `src/test/test_omnicore`
- `src/qt/omnicore-qt`
- `src/qt/test/test_omnicore-qt`

The installer for Windows is currently build with the file name `omnicore-0.0.10.0-dev-win64-setup.exe`, and this is based on the values used in the configuration file.

When using the Windows installer, the following files are created or copied to the install destination:

- `daemon\omnicored.exe`
- `daemon\omnicore-cli.exe`
- `doc\*.*`
- `copying.txt`
- `omnicore-qt.exe`
- `readme.txt`
- `release-notes.txt`
- `uninstall.exe`

This is similar to the original Bitcoin Core behavior, except that it uses Omni Core related documents.

On OS X it *should* be `OmniCore-Qt.app`. I haven't tested it yet, but I also haven't touched that part in the recent update of this PR.

**Comment by msgilligan** [comment]
I submitted a PR a while back with Mac OS X rebranding (Bitcoin -> 'Master Core') so I can check out the Mac OS build process for the same issues.

**Comment by dexX7** [comment]
This would be really great!

In these files are the Mac specific changes:

- [Makefile.am](https://github.com/OmniLayer/omnicore/pull/91/files#diff-c949f93d03f44a4217d7a138f9e2e54a)
- [contrib/macdeploy/fancy.plist](https://github.com/OmniLayer/omnicore/pull/91/files#diff-c5ee10fdc4e723d0e3bc04985a9b0a88)
- [contrib/macdeploy/macdeployqtplus](https://github.com/OmniLayer/omnicore/pull/91/files#diff-e2c1caf49768933aa4719737154e63c8)
- [share/qt/Info.plist.in](https://github.com/OmniLayer/omnicore/pull/91/files#diff-5fdfe8cc3c56a1deb4896e4532024326)

When comparing it with https://github.com/mastercoin-MSC/mastercore/pull/203, it seems to add up.

I did not change the `CFBundleIdentifier` in `Info.plist.in` though. According to [developer.apple.com](https://developer.apple.com/library/prerelease/ios/documentation/General/Reference/InfoPlistKeyReference/Articles/CoreFoundationKeys.html#//apple_ref/doc/uid/20001431-102070):

> CFBundleIdentifier (String - iOS, OS X) uniquely identifies the bundle. Each distinct app or bundle on the system must have a unique bundle ID. The system uses this string to identify your app in many ways. For example, **the preferences system uses this string to identify the app for which a given preference applies**;

We *should* use an unqiue identifier, but I wasn't sure, if this has an impact on Bitcoin Core settings, i.e. "ideally a former Bitcoin Core user should be able to switch to Omni Core and back with ease".

**Comment by zathras-crypto** [comment]
> On OS X it should be OmniCore-Qt.app. I haven't tested it yet

The name is correct but the icon is still the Bitcoin Core icon:
screenshot from 2015-07-02 12 02 59

Interestingly the dock icon starts with the Bitcoin Core icon but changes to the correct icon during initialization:
screenshot from 2015-07-02 12 06 52   screenshot from 2015-07-02 12 01 52

**Comment by dexX7** [comment]
Hehe interesting.

We have the following images to update:

- Installer: [contrib/macdeploy/background.png](https://github.com/OmniLayer/omnicore/blob/omnicore-0.0.10/contrib/macdeploy/background.png) (there is also a `.psd`)

I haven't found a reference in the OS X files to pin down the icon, but it's probably one of the following:

- pixmaps/bitcoin.ico (via share/ui.rc)
- src/qt/res/src/bitcoin.svg
- share/pixmaps

**Comment by dexX7** [comment]
@zathras-crypto: do you have any higher resolution Omni images? Or could you maybe even replace `bitcoin.icns`? :) Other than that, I think this PR is ready.

**Comment by zathras-crypto** [comment]
> do you have any higher resolution Omni images?

I do but not with transparent backgrounds to use for a higher res icon :(  I'll arrange to get some originals but for PR https://github.com/OmniLayer/omnicore/pull/114 should do the trick.

> Other than that, I think this PR is ready.

Agreed, looks good to me :)
